### PR TITLE
Add conf-m4, conf-ncurses, conf-perl, conf-which and conf-zlib

### PR DESCRIPTION
**Pull request by timbertson**
Also added `nixpkgs` depexts to some existing conf-* packages

This should make #5338 easier.

### CODE REVIEW

**Comment by OCamlPro-Henry** [comment]
There is a missing dependency towards `debianutils` in the `depext` of `conf-which`. That why the CI failed.

**Comment by timbertson** [comment]
Thansk @OCamlPro-Henry, it's quite hard to google for "what provides which" ;). The travis build still seems to be failing on `conf-which`, I'm not really sure why though. Any ideas?

File: packages/conf-which/conf-which.1/opam [hunk 1]
```diff
@@ -0,0 +1,12 @@
+opam-version: "1.2"
+maintainer: "tim@gfxmonk.net"
+homepage: "http://www.gnu.org/software/which/"
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+dev-repo: "https://github.com/ocaml/opam-repository.git"
+license: "GPL-2+"
+build: [["which" "--version"]]
```
Comment (dbuenzli) [inline packages/conf-which/conf-which.1/opam]: According to the build logs `which` does not support a `--version` option.

**Comment by OCamlPro-Henry** [comment]
There is no `--version` argument in this `which`. You might try `which which` ?

**Comment by timbertson** [comment]
`which which` it is! (funny, mine has `--version` on fedora). Now only the OSX build is failing - does that matter for this change? AFAIK existing packages which use nurses would fail there too, although I haven't tested.

**Comment by dsheets** [comment]
@timbertson, the opam package `curses` builds fine on OS X with no further changes as (some version of) `ncurses` is (currently) shipped with OS X. I believe the `conf-ncurses` packages should only run the `pkg-config` check if `{ os != "darwin" }`.

**Comment by timbertson** [comment]
Thanks @dsheets, looks like it should be ready to merge now :)

**Comment by UnixJunkie** [comment]
conf-which, really ?! I never logged into a unix-like with no which command.
I even suspect many startup scripts would not work if this command is not available.

**Comment by OCamlPro-Henry** [comment]
@UnixJunkie Yeah, really ! It seems required by NixOS. Not tested myself though, and I wouldn't have imagine this might miss without this PR.

**Comment by timbertson** [comment]
Yep, the default nix build environment includes very little - not even
which.

**Pull request merged**